TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -197,37 +215,92 @@ private void runReplayWithIOStreams(Duration observedPacketConnectionTimeout,
             throw e;
         } finally {
             trafficToHttpTransactionAccumulator.close();
-            var allRemainingWorkArray = requestToFinalWorkFuturesMap.values().toArray(CompletableFuture[]::new);
-            log.info("All remaining work to wait on: " +
-                    Arrays.stream(allRemainingWorkArray).map(cf->cf.toString()).collect(Collectors.joining()));
-            // remember, this block is ONLY for the leftover items.  Lots of other items have been processed
-            // and were removed from the live map (hopefully)
-            CompletableFuture.allOf(allRemainingWorkArray)
-                    .handle((t, v) -> {
-                        log.info("stopping packetHandlerFactory's group");
-                        packetHandlerFactory.stopGroup();
-                        return null; // squash exceptions for individual requests
-                    })
-                    .get();
+            var PRIMARY_LOG_LEVEL = Level.INFO;
+            var SECONDARY_LOG_LEVEL = Level.INFO;
+            var logLevel = PRIMARY_LOG_LEVEL;
+            for (var timeout = Duration.ofSeconds(1); ; timeout = timeout.multipliedBy(2)) {
```

Comment:
This is hard to follow, not sure what we are looping here

Reply:
We're running waitForRemainingWork repeatedly.  First we do it with a 1 second timeout, then we double it indefinitely.  We'll also change the log level that we want to report what's still remaining after the first run - the thought being that after the first second or whatever the wait duration was, "normal" things should be wrapped up pretty well.

Reply:
Ahhh - gotcha.. You're right.  Thanks.